src/lib/hash.ts

```diff
@@ -43,6 +48,34 @@ const Poseidon = {
     return Poseidon_.hash(input, isChecked);
   },
 
+  hashToGroup(input: Field[]) {
+    let isChecked = !input.every((x) => x.isConstant());
+    // y = sqrt(y^2)
+    let { x, y } = Poseidon_.hashToGroup(input, isChecked);
+
+    let { x0, x1 } = Circuit.witness(provable({ x0: Field, x1: Field }), () => {
+      // the even root of y^2 will become x0, so the APIs are uniform
+      let isEven = y.toBigInt() % 2n === 0n;
+
+      // r is the second root of sqrt(y^2)
+      let r = y.mul(-1);
```

Comment:
Could we factor that computation, albeit not costly with the `let y_ = y.mul(-1)` below?